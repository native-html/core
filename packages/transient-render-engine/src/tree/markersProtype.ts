import { Markers } from './tree-types';

type MarkersShape = Omit<Markers, 'extends'>;

interface TNodeDescriptor {
  tagName: string | null;
  classes: string[];
  id: string | null;
  attributes: Record<string, string>;
}

const markersProtype: Markers = {
  anchor: false,
  edits: 'none',
  lang: 'en',
  olNestLevel: -1,
  ulNestLevel: -1,
  direction: 'ltr',
  extend(this: Markers) {
    return Object.create(this);
  },
  toString() {
    let print = 'Markers {\n';
    for (const key in this) {
      //@ts-ignore
      const val = this[key];
      if (typeof val !== 'function') {
        print += `  ${key}: ${JSON.stringify(val)}\n`;
      }
    }
    return print + '}';
  }
};

export default markersProtype;
