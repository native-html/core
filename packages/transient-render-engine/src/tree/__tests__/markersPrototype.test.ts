import markersProtype from '../markersProtype';

describe('markersPrototype', () => {
  describe('toString', () => {
    it('should provide a string representation of own and inherited properties', () => {
      const markers = markersProtype.extend();
      markers.anchor = true;
      expect(markers.toString()).toMatchSnapshot();
    });
  });
});
