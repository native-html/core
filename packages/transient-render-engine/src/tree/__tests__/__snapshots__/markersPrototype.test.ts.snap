// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markersPrototype toString should provide a string representation of own and inherited properties 1`] = `
"Markers {
  anchor: true
  edits: \\"none\\"
  lang: \\"en\\"
  olNestLevel: -1
  ulNestLevel: -1
  direction: \\"ltr\\"
}"
`;
